File: examples/sdk/python/3_validation.py
# /// script
# dependencies = ["envilder>=0.3.1"]
# ///

# Secret validation: fail fast if any secret is missing or empty
from envilder import Envilder, validate_secrets

secrets = Envilder.resolve_file("../../../envilder.json")
validate_secrets(secrets)  # raises SecretValidationError if any value is empty

for key, value in secrets.items():
    print(f"{key} = {value}")

File: examples/sdk/nodejs/3_validation.ts
// Secret validation: fail fast if any secret is missing or empty
import { Envilder, validateSecrets } from '@envilder/sdk';

const secrets = await Envilder.resolveFile('../../../secrets-map.json');
validateSecrets(secrets); // throws SecretValidationError if any value is empty

for (const [key, value] of secrets) {
  console.log(`${key} = ${value}`);
}
